src/wasm/wasm-type.cpp

```diff
@@ -1137,6 +1207,9 @@ std::ostream& TypePrinter::print(Type type) {
   }
 
   return printChild(type, [&]() {
+    if (isTemp(type)) {
```

Comment:
with these uses of `isTemp` I think we can remove the hacks on lines 278-281 and 287-290.

Reply:
Oh hm, I thought I had already done that. Thanks for the catch!